OpenRA.Mods.Common/Traits/World/ResourceLayer.cs

```diff
@@ -177,6 +213,45 @@ public void TickRender(WorldRenderer wr, Actor self)
 				dirty.Remove(r);
 		}
 
+		public void Tick(Actor self)
+		{
+			if (--ticks > 0)
```

Comment:
This segment should be rewired now that you're using actual ticks and not a value related to the default gamespeed.